plugins/e57/io/E57Writer.cpp

```diff
@@ -288,23 +303,23 @@ void E57Writer::setupWriter()
         {
             auto bounds = e57plugin::getPdalBounds(pdal::Dimension::Id::Red);
             proto.set(e57Dimension,
-                e57::IntegerNode(*m_imageFile, 0, bounds.first,bounds.second));
+                      e57::IntegerNode(*m_imageFile, 0, bounds.first,bounds.second));
```

Comment:
These should be using the colour bounds you've found while writing.

Reply:
No. We should not do that.
`setupWriter()` is called in `ready()`, At the time color and intensity limits are at default state i.e. 255 and 1 respectively.
And if `ChunkWriter::write()` founds value grater than that, it will definately throw an out of range exception.

Reply:
Good